### PR TITLE
NSFS | NC | NC Coretests changes - PR 2/3

### PR DESCRIPTION
**Pull request by romayalon**
WIP - This PR is dependent on merging https://github.com/noobaa/noobaa-core/pull/7719
### Explain the changes
1. Makefile - added run-nc-tests makefile rule for running nc_index.js tests file.
2. test_utils.js - added a helper function for get_coretest_path() that chooses coretest/nc_coretest path by process.env.NC_CORETEST.
3. nc_coretest.js - config_dir, storage, and other setups and starting nsfs.js process, added mapping between rpc_client calls and manage_nsfs commands or other mocks.
4. nc_index.js - added NSFS relevant test files.
5. test_bucketspace.js - added skips by process.env.NC_CORETEST of non-relevant tests, and different error_codes.

### Issues: Fixed #xxx / Gap #xxx
1. Gap - nc_index.js contains commented test files that needs to be added as well.
2. github action for running nc_index.js

### Testing Instructions:
1. 

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/test/unit_tests/nc_coretest.js [hunk 1]
```diff
@@ -0,0 +1,450 @@
...
+    const list = await exec_manage_cli(nc_nsfs_manage_entity_types.ACCOUNT, nc_nsfs_manage_actions.LIST, options);
+    return { accounts: JSON.parse(list).response.reply };
+}
+
+/**
+ * update_account_s3_access_manage updates an account using manage_nsfs
+ * @param {object} options
+ * @returns {Promise<void>}
+ */
+async function update_account_s3_access_manage(options) {
```
Comment (naveenpaul1) [inline src/test/unit_tests/nc_coretest.js:327]: @romayalon are we validating this account update anywhere else? just a doubt

**Review by naveenpaul1 — COMMENTED** [review]
A comman doubt : - How this different from the normal test?

File: src/test/unit_tests/nc_coretest.js (hunk 1 above)
Comment (romayalon) [reply src/test/unit_tests/nc_coretest.js:327]: on the tests, this function equal to the rpc function - update_account_s3_access(), the difference here is that on Non containerized version we don't have RPC server so we do the update by running manage NSFS cli calls instead and return rpc like response.

**Comment by romayalon** [comment]
@naveenpaul1 The regular Coretest runs noobaa endpoint (and other components of noobaa core) and then a test can run full s3 flow.
NC Coretest runs nsfs.js instead of running the endpoint and by that, we run the NC code. Then a test can run full s3 flow on NC environment. I hope it makes it clearer, let me know if there are more questions.

**Review by naveenpaul1 — APPROVED** [review]
LGTM

**Pull request merged**